src/irmin/irmin.mli

```diff
@@ -3417,26 +3417,39 @@ module type SYNC = sig
   val fetch : db -> ?depth:int -> remote -> (commit, fetch_error) result Lwt.t
   (** [fetch t ?depth r] populate the local store [t] with objects for
       the remote store [r], using [t]'s current branch. The [depth]
-      parameter limits the history depth. Return [None] if either the
-      local or remote store do not have a valid head. *)
+      parameter limits the history depth.
+
+      If the remote repository has no commits then [fetch] will return
+      [Error `No_head]. In the case where calling [fetch] on an empty repository
+      is expected you should explicitly handle [Error `No_head]. *)
```

Comment:
I'm still curious when `` `Msg of string `` is returned, and what `` `Not_available `` specifically means (is the remote not reachable? does the repository not exist on the remote? are the provided credentials not sufficient for fetching?)

Reply:
I wasn't sure what `` `Not_available `` myself - turns out it is only used when an Irmin store is not configured to use `fetch`/`push`/`pull`.

Reply:
in the end, it's __your__ (well, or @samoht) API design -- but I'd _either_ make a variant with the specific failures _or_ use a `` `Msg of string`` -- the first I'd do if I expect API clients to handle specific errors, the second if the only handling is to print a warning/error to the user. but mixing the two is confusing for me. i also don't understand `irmin not configured for fetching` (and in which scenario I'll encounter this), but maybe that's just me.

Reply:
I agree, it seems like something like:

```ocaml
val fetch :
    db ->
    ?depth:int ->
    remote ->
    (commit option, [ `Msg of string ]) result Lwt.t
```

makes the most sense. I will create a new PR to remove the unintuitive error states.